torch/utils/checkpoint.py

```diff
@@ -35,6 +35,34 @@ def check_backward_validity(inputs: Iterable[Any]) -> None:
         warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
 
 
+def _get_device_module(device='cuda'):
+    device_module = getattr(torch, device)
+    return device_module
+
+class _DefaultDevice(object):
+    _default_device_type = "cuda"
+
+    @staticmethod
+    def set_device_type(device: str = "cuda"):
+        _DefaultDevice._default_device_type = device
+
+    @staticmethod
+    def get_device_type():
+        return _DefaultDevice._default_device_type
+
+def _infer_device_type(*args):
+    device_types = list({arg.device.type for arg in args
+                        if isinstance(arg, torch.Tensor) and not arg.device.type == "cpu"})
+    if len(device_types) > 1:
+        warnings.warn("Tensor args except CPU tensor are on at least two devices ", device_types,
```

Comment:
Maybe something along the lines of: 
```
Tensor arguments, excluding CPU tensors, are detected on at least two devices.
When a CUDA device is among them, it will be prioritized; otherwise, the first
device encountered will be used. Keep in mind that the device state will only
be saved for one device, even though tensors are present on multiple devices.
Consequently, if any checkpointed functions involve randomness, this may result
in incorrect gradients.
```